File: client/src/app/models/statsDto.ts
 
export class StatsDto {
  productCount: number;
  brandCount: number;
  userCount: number;
  roleCount: number;
  dataEnterBarChart: ChartSeries[];
  productByBrandTypePieChart: { [key: string]: number } | null;

  constructor() {
      this.productCount = 0;
      this.brandCount = 0;
      this.userCount = 0;
      this.roleCount = 0;
      this.dataEnterBarChart = [];
      this.productByBrandTypePieChart = null;
  }
}

// ChartSeries sınıfının TypeScript karşılığı
class ChartSeries {
  name?: string;
  data?: number[];

  constructor(name?: string, data?: number[]) {
      this.name = name;
      this.data = data;
  }
}

File: Server/Server/src/Core/Application/Dashboard/StatsDto.cs
namespace Server.Application.Dashboard;

public class StatsDto
{
    public int HastaCount { get; set; }
    public int BasvuruCount { get; set; }
    public int PersonelCount { get; set; }
    public int UserCount { get; set; }
    public int RoleCount { get; set; }
    public List<ChartSeries> DataEnterBarChart { get; set; } = new();
}

public class ChartSeries
{
    public string? Name { get; set; }
    public double[]? Data { get; set; }
}
